Doppler.Sap/Factory/CreateOrUpdateBusinessPartnerHandler.cs

```diff
@@ -31,6 +31,46 @@ public async Task<SapTaskResult> Handle(SapTask dequeuedTask)
         {
             dequeuedTask = await _sapTaskHandler.CreateBusinessPartnerFromDopplerUser(dequeuedTask);
 
+            if (string.IsNullOrEmpty(dequeuedTask.ExistentBusinessPartner.FederalTaxID))
+            {
+                return await CreateBusinessPartner(dequeuedTask);
+            }
+            else
+            {
+                return await UpdateBusinessPartner(dequeuedTask);
+            }
```

Comment:
Are you sure of this? How could you update or create a business partner without a FederalTaxId?

Reply:
Nevermind this. I just saw that in the dequeuedTask you have both the existent business partner and the new one. 👍

Reply:
This is the same that we have in old SAP service, there is into the controller, here I move it the logic to task handler. 
If the ExistentBusinessPartner dont have FederalTaxID it's because is a new BusinessPartner. If has FederalTaxID its because is an Update.
Maybe Im not understand you?